portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,30 +38,46 @@
 public class DataServlet extends HttpServlet {
   private final String NAME = "name";
   private final String GET_LOAD_COMMENT_QUANTITY = "loadcommentquantity";
+  private final String USER_EMAIL = "email"; 
 
   public static final String COMMENT = "comment";
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    //server side Get check: login required to load comments
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/login_out");
+      return;
+    }
+
+    //user specified comments load quantity should be a number
     int maxCommentsLoad = getUserInput(request);
     if (maxCommentsLoad == -1) {
       response.setContentType("text/html");
       response.getWriter().println("Please enter an integer.");
       return;
     }
 
-    Query query = new Query(COMMENT).addSort("timestamp", SortDirection.DESCENDING);
+    String userEmail = userService.getCurrentUser().getEmail();
 
+    /**search for comments made with the current user's email address
+      start with last comment made
+    */
+    Query query = new Query(COMMENT).setFilter(new Query.FilterPredicate(USER_EMAIL, Query.FilterOperator.EQUAL, userEmail)).addSort("timestamp", SortDirection.DESCENDING);
```

Comment:
Please break this line. You could refer to the style guide for maximum characters per line allowed (if I recall correctly, it's 100).

Reply:
done